api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -251,10 +251,6 @@ private FirecloudWorkspaceId generateFirecloudWorkspaceId(String namespace, Stri
       Workspace workspace) {
     org.pmiops.workbench.db.model.Workspace dbWorkspace = getDbWorkspaceCheckExists(
         workspaceNamespace, workspaceId);
-    if (workspace.getDataAccessLevel() != null) {
-      dbWorkspace.setDataAccessLevel(
-          DataAccessLevel.fromValue(workspace.getDataAccessLevel().name()));
```

Comment:
This looks like it's converting enum => string => enum. I think it's because we've declared separate enums (which *should* have the same strings in them). At https://github.com/all-of-us/workbench/blob/03aa50f32fe4ff4138e47808c99562e6fa01c5bf/api/src/main/resources/workbench.yaml#L642 you can see Swagger's syntax `*DATA_ACCESS_LEVEL` which reuses values but doesn't reuse the Java enum.

It may simplify things if you switch to an object reference, like https://github.com/all-of-us/workbench/blob/03aa50f32fe4ff4138e47808c99562e6fa01c5bf/api/src/main/resources/workbench.yaml#L831 . It won't be a direct mapping, since for `Authority` I needed an array of enum values (not a single value), but it should be pretty similar.

I don't know why the `fromValue` is failing now (especially after the null check), but hopefully using the same type throughout would make it Just Work.

Reply:
I guess a big question in my mind is whether we need to do this in the first place. I talked to @jmandel on slack, and he seemed to think if we have a mutable Data Access Level, it should be "ratcheted" (Can increase but not decrease), but it may be more straightforward to have users pick at the time of Workspace creation.

Reply:
Hmm. It's weird that it doesn't just reuse the same enum here. I will look into whether there's a way to do this in Swagger.

Anyway, removing this for now seems fine.

Reply:
Locking the access level seems OK to me, though if we know we want it to ratchet up we may as well nail that down now.

Either way, we should explicitly reject the request if it tries to make an edit that we won't support. (And, unfortunately, I think that requires making the data access level comparison work -- but being able to compare the value from Swagger and our db is a good thing in any case.)

Reply:
@danqrodney Yes, you can have Swagger reuse enums using object references (`$ref: "#/definitions/MyEnum"`) -- I set that up for `Authority` in https://github.com/all-of-us/workbench/pull/171/files .

Reply:
Cool. I will take an AI to update the API to use DataAccessLevel everywhere.

Reply:
I should be using it everywhere now. If errors pop up, they aren't hard to fix.